src/scripts/main.js

```diff
@@ -1,3 +1,217 @@
 'use strict';
 
 // write your code here
+const button = document.querySelector('.button');
+const messeageStart = document.querySelector('.message-start');
+const messageWin = document.querySelector('.message-win');
+const messageLose = document.querySelector('.message-lose');
+const gameScore = document.querySelector('.game-score');
+const rows = document.querySelectorAll('.field-row');
+let score = 0;
+let gameField = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+function setGameField() {
+  for (let row = 0; row < 4; row++) {
+    for (let cell = 0; cell < 4; cell++) {
+      rows[row].children[cell].textContent = gameField[row][cell] || '';
+
+      rows[row].children[cell].className
+        = `field-cell field-cell--${gameField[row][cell]}`;
+
+      if (gameField[row][cell] === 2048) {
+        messageWin.classList.remove('hidden');
+      };
+    }
+  }
+};
+
+function hasEmptyCell() {
+  for (let row = 0; row < 4; row++) {
+    for (let cell = 0; cell < 4; cell++) {
+      if (gameField[row][cell] === 0) {
+        return true;
+      }
+    }
+  }
+
+  return false;
+}
+
+function canMove() {
+  for (let r = 0; r < 4; r++) {
```

Comment:
It's better to write full word instead of one letter as it can be unclear. Moreover, you've already wrote two functions with full words and it's good to keep same logic for variables with same meaning.
```suggestion
  for (let row = 0; row < 4; row++) {
```